src/Test.EndToEnd.Baselining/BaseliningDetailLogger.cs

```diff
@@ -62,7 +69,8 @@ public void Write(SarifLog newBaselineLog, SarifLog baselineLog, SarifLog curren
                     case BaselineState.Unchanged:
                     case BaselineState.Updated:
                         // Find and write old Result from previous Baseline (to get pre-merged properties)
-                        Result previousResult = baselineLog.FindByGuid(result.CorrelationGuid);
+                        Result previousResult = null;
+                        baselineResultsByGuid.TryGetValue(result.CorrelationGuid, out previousResult);
```

Comment:
>TryGetValue [](start = 46, length = 11)

Ultra-nit: Now that you're using `TryGetValue`, it's more idiomatic to test on its return value:
```
if (!baselineResultsByGuid.TryGetValue(result.CorrelationGuid, out Result previousResult)
```

Reply:
Left alone - the reason I wrote this as a statement is that I want to be clear the dictionary lookup is a step, and not a side-effect free conditional. I personally find it confusing when conditional expressions have side effects which the later code requires, so re-ordering the conditional blocks (for example) will cause bugs.

When it's a single condition if statement (if (!Map.TryGetValue(key, out value)) { value = new().. }) I tend to leave them inline, but when there are multiple conditions I don't like them. That's the rationale, anyway.